src/PactNet.Abstractions/Verifier/ProviderState/IStateHandler.cs

```diff
@@ -0,0 +1,30 @@
+using System.Collections.Generic;
+
+namespace PactNet.Verifier.ProviderState
+{
+    /// <summary>
+    /// Defines the state handler interface
+    /// </summary>
+    public interface IStateHandler
+    {
+        /// <summary>
+        /// The provider state description
+        /// </summary>
+        string Description { get; }
+
+        /// <summary>
+        /// When the provider state is executed
+        /// </summary>
+        StateAction Action { get; }
+
+        /// <summary>
+        /// Invoke a provider state
+        /// </summary>
+        void Execute();
+
+        /// <summary>
+        /// Invoke a provider state
+        /// </summary>
+        void Execute(IDictionary<string, string> args);
```

Comment:
I think it the reason why splitting this into an interface with and without args is best - you can invoke this method on a handler that doesn't take arguments, or likewise you can invoke the non-argument method on a handler that requires arguments, and then it's unclear what should happen.

Reply:
Yes, I went with that design first but was stuck on the execution end. I don't remember for what reason but the final design doesn't convince me either. It works at the moment but needs refining.